src/main/java/cubicchunks/server/ServerCubeCache.java

```diff
@@ -212,15 +214,16 @@ public boolean unloadQueuedChunks() {
 		final int MaxNumToUnload = 400;
 
 		// unload cubes
-		for (int i = 0; i < MaxNumToUnload && !this.cubesToUnload.isEmpty(); i++) {
-			long cubeAddress = this.cubesToUnload.poll();
+		Iterator<CubeCoords> iter = this.cubesToUnload.iterator();
+		while (iter.hasNext()) {
+			CubeCoords coords = iter.next();
```

Comment:
you never limit the amount of unloaded cubes per tick. `unloadQueuedChunks` should be really called `tick`.